pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -852,14 +852,16 @@ impl RemoteTimelineClient {
         let remaining: Vec<RemotePath> = remaining
             .into_iter()
             .filter(|p| p.object_name() != Some(IndexPart::FILE_NAME))
+            .inspect(|path| {
+                if let Some(name) = path.object_name() {
+                    info!(%name, "deleting a file unbound to index_part.json");
+                } else {
+                    warn!(%path, "deleting a nameless or non-utf8 object unbound to index_part.json");
```

Comment:
We shouldn't in general have non-utf8 paths anywhere, like here they scare me because it's not that the object has no name, it could also be that it fails to convert to str.

Instead of PathBuf, we should just use https://docs.rs/camino/latest/camino/.

Reply:
Issue created (finally): #4689